spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/EAMSGD.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.optim
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{T, Table}
+
+import scala.reflect.ClassTag
+
+/**
+ * An implementation of EASGD/EAMSGD.
+ * EAMSGD is based on the Nesterov's momentum scheme,
+ * When mom==0, it is the EASGD.
+ * See the original paper: https://cs.nyu.edu/~zsx/nips2015.pdf
+ * @param ev numeric operator
+ * @tparam T data type
+ */
+class EAMSGD[@specialized(Float, Double) T: ClassTag](implicit ev: TensorNumeric[T])
+  extends OptimMethod[T] {
+
+  /**
+   * Elastic Averaging (Momentum) SGD implementation
+   *
+   * @param feval   a function that takes a single input (X), the point of a evaluation,
+   *                and returns f(X) and df/dX
+   * @param x       the initial point
+   * @param config  a table with configuration parameters for the optimizer
+   *                config("learningRate"): learning rate
+   *                config("learningRateDecay"): learning rate decay
+   *                config("learningRateDecayPower"): learning rate decay power
+   *                config("weightDecay"): weight decay
+   *                config("momentum"): momentum
+   *                config("movingRate"): moving rate \alpha
+   *                config("commPeriod"): sync update (communication period) \atu
+   * @param state   a table describing the state of the optimizer; after each call the state
+   *                is modified
+   *                state("vt"): vector of movement at time t
+   *                state("suw"): vector of weight for communication with the center variable
+   *                state("sug"): vector of gradient for communication with the center variable
+   * @return the new x 1D vector and the function list, evaluated before the update
+   */
+  override def optimize(feval: (Tensor[T]) => (T, Tensor[T]), x: Tensor[T],
+    config: Table, state: Table = null): (Tensor[T], Array[T]) = {
```

Comment:
No.
Indeed currently we simply implement a **Synchronous** EASGD and the update logic for Synchronous one is simple and was moved to `DistOptimizer`. 
This `EAMSGD.optimzie` method is planned for **Asynchronous** EASGD which is not implemented now.